docs/blog/posts/multi-stage-delivery-using-gitops.md

```diff
@@ -0,0 +1,413 @@
+  The deployment traces are gathered by the OpenTelemetry Collector and forwarded to Jaeger,
+  where we can view the generated traces.
+- [Prometheus](https://prometheus.io): Provides monitoring data for the application.
+
+Note that for this blog post, we are assuming that these tools are already
+installed on the Kubernetes cluster, as going through the installation
+of each of those would exceed the scope of this blog post.
+
+As this is quite a complex setup, here is a preview of what steps
+are involved in setting everything up.
+We are going to:
+
+1. Set up the environment by:
+     1. Setting up a GitHub repository with an access token, GitHub, workflows, and GitHub actions
+     1. Preparing a Kubernetes namespace for each stage (`dev` and `prod`)
+     1. Preparing the ArgoCD applications with an appropriate Helm chart for each
+     1. Applying labels to associate the Deployment resource with the KeptnWorkload resource
+     1. Defining Keptn pre-/post-deployment checks and tasks
+     1. Defining the metadata to be passed through the deployment traces
+     1. Define a TraceParent that links the deployment traces of the `prod` stage to those of the `dev` stage
+1. Run the promotion flow by:
+     1. Creating a pull request to update our `dev` environment
+     1. Merging the automatically created pull request to promote the updated version into `prod`
+     1. Inspecting the generated deployment traces for both stages
+   and see how they are connected with each other
+
+## Setting up the Environment
+
+Now it's time to set up our environment and connect all the tools mentioned above
+with each other.
+
+### Set up the GitHub repository
+
+First things first, since we talk about GitOps in this article, we need
+a git repository to host the Helm chart of our application.
+In this example, we are using GitHub, since this also allows us
+to make use of GitHub Actions to implement the promotion from
+`dev` to `production`.
+
+In this example, we are using [this repository](https://github.com/bacherfl/keptn-analysis-demo)
+as an upstream repository.
+If you would like to try the demo yourself, feel free to fork this repository
+and start experimenting with Keptn from there.
+
+#### Create personal access token
+
+The first step in setting up the demo environment is to
+create a personal access token for accessing the GitHub API.
+This token will be used by the container running the promotion
+task during the post-deployment phase of the `KeptnApp` within
+the `dev` stage.
+Using the access token, the container triggers a GitHub action
+that creates a pull request to promote the version that has
+been deployed in `dev` into `production`.
+Using this GitHub actions-based approach instead of directly
+interacting with the git repository in the container executing the promotion
+step has the advantage of not having to grant the container any write permissions to the
+repository.
```

Comment:
```suggestion
By using GitHub actions rather than interacting directly with the Git repository in the container that executes the promotion
step lets us avoid granting the container any write permissions to the
repository.
```